File: rgctl-tests/ecommerce-go/internal/pkg/password/password.go
package password

import "golang.org/x/crypto/bcrypt"

func Hash(plain string) (string, error) {
	hash, err := bcrypt.GenerateFromPassword([]byte(plain), bcrypt.DefaultCost)
	if err != nil {
		return "", err
	}
	return string(hash), nil
}

func Verify(plain, hash string) bool {
	return bcrypt.CompareHashAndPassword([]byte(hash), []byte(plain)) == nil
}

File: rgctl-tests/ecommerce-rust/src/utils/password.rs
use crate::error::{AppError, AppResult};

pub fn hash(password: &str) -> AppResult<String> {
    bcrypt::hash(password, bcrypt::DEFAULT_COST).map_err(|e| AppError::BadRequest(e.to_string()))
}

pub fn verify(password: &str, hash: &str) -> AppResult<bool> {
    bcrypt::verify(password, hash).map_err(|e| AppError::BadRequest(e.to_string()))
}
